Wiki thread lighting app (#28)
canonical/openthread-border-router-snap: wiki-thread-lighting-app.md

## How to Set Up and Control a Matter Thread Lighting Device

This guide will walk you through the process of building, running, and controlling a Matter Thread lighting application over a Thread network. 

Here are step-by-step instructions:

#### Hardware Setup
- Machine A
  - nRF52840 dongle as [Radio Co-Processor (RCP)](https://openthread.io/platforms/co-processor#radio_co-processor_rcp)
  - Choose one of the following options:

>     - PC running Ubuntu Classic 23.04 OS

Author: I was referring to classic Ubuntu OS variants, such as Ubuntu Server and Ubuntu Desktop. It has now been updated to the one I tested, which is Ubuntu Desktop.

>     - PC running Ubuntu Classic 23.04 OS

Reviewer: You could say "Ubuntu Server/Desktop". There is no distro named "Ubuntu Classic"; Canonical refers to non-Core Ubuntu variants as the [classic](https://www.merriam-webster.com/dictionary/classic) Ubuntu.

>   - Choose one of the following options:
>     - PC running Ubuntu Classic 23.04 OS
>     - Raspberry Pi 4B running Ubuntu Server 22.04 OS or Ubuntu Core 22, and an LED light

Reviewer: > Choose one of the following options

A little complicated. Is this implying that Ubuntu Server/Desktop 23.04 on Pi isn't going to work? What about UC22 on a PC?

Maybe the Option 1 and Option 2 referenced in the instructions should also be used here.

```suggestion
  - Option 1: PC running Ubuntu Classic 23.04 OS
  - Option 2: Raspberry Pi 4B running Ubuntu Server 22.04 OS or Ubuntu Core 22, and an LED light
```

- Machine B
  - nRF52840 dongle as RCP
  - PC running Ubuntu Classic 23.04 OS
 
#### Environment Setup
- Machine A running
  - [OpenThread Border Router (OTBR) snap](https://snapcraft.io/openthread-border-router)
  - Matter Thread ligting App
- Machine B running
  - OTBR snap
  - [Chip Tool snap](https://snapcraft.io/chip-tool)

#### Versions Tested in This Guide
- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)

#### 1. Form a Thread Network on Machine A

> On Machine A, following this guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 

Reviewer: ```suggestion
On Machine A, following the guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
```

and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 

> pelase refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).

Reviewer: ```suggestion
please refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```

#### 2. Run a Matter Thread Lighting App on Machine B
There are two options to run a Matter Thread lighting application. 

> **Option 1**: Native Build Lighting App

Reviewer: ```suggestion
**Option 1**: Native Built Lighting App
```

> You can choose this option to build and run the Matter Thread lighting app from the Matter SDK in the native shell. With this option, you can observe the lighting status (on/off) from the logs.

Reviewer: ```suggestion
You can choose this option to build and run the [Matter Thread lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. With this option, you can observe the lighting status (on/off) from the logs.
```

To get started, follow these steps:

- 1. Install the Matter SDK with a shallow clone for the Linux platform:
```
git clone https://github.com/project-chip/connectedhomeip.git --depth=1
cd connectedhomeip
git checkout 6b01cb977127eb8547ce66d5b627061dc2dd6c90
scripts/checkout_submodules.py --shallow --platform linux
```
- 2. Build the lighting app:
```
cd examples/lighting-app/linux
gn gen out/debug
ninja -C out/debug
```
- 3. Run the lighting app with Thread feature enabled:
```
./out/debug/chip-lighting-app --thread
```

**Option 2**: GPIO Commander Snap

Alternatively, you can use this option to set up the Matter Thread lighting application by installing the Matter Pi GPIO Commander snap. 
This turns your Raspberry Pi into a Matter Thread lighting device, allowing you to control the connected LED, turning it on or off.

Follow the instructions for installing the matter-pi-gpio-commander snap, 
provided at [Setup OpenThread Border Router with nRF52840 Dongle](https://github.com/canonical/matter-pi-gpio-commander/wiki/Setup-and-control-a-lighting-device#installation).

> TBA: Then, running the GPIO commander snap with Thread feature enabled.

Reviewer: Also, need to set the `--thread` flag via the snap option.

Author: Indeed, this information regarding the --thread flag could be included on the GPIO Commander Snap's side. And a link could be simply added here to direct users to it.

#### 3. Run OTBR D-Bus Daemon on Machine B
Check the [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) section

> to run OTBR dbus daemon, which is needed to support lighting app's Thread feature as pointed 

Reviewer: The dbus daemon runs on the system. What is needed here is to run the OTBR agent. GUI is optional.

The lighting app communicates with OTBR agent via the dbus message bus.

Follow the instructions for installing the matter-pi-gpio-commander snap, 
provided at [Setup OpenThread Border Router with nRF52840 Dongle](https://github.com/canonical/matter-pi-gpio-commander/wiki/Setup-and-control-a-lighting-device#installation).

TBA: Then, running the GPIO commander snap with Thread feature enabled.

#### 3. Run OpenThread Border Router Agent on Machine B
Check the [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) section
to run OTBR agent. The lighting app communicates with OTBR agent via the dbus message bus. This is necessary to support the lighting app's Thread feature, as explained
[here](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux#commandline-arguments).

> Note: Only the OTBR agent is required, and the GUI (openthread-border-router.otbr-web) is optional. The OTBR agent can be started by running the following command:
> ```bash
> sudo snap start openthread-border-router.otbr-agent

> > ```

Reviewer: The link on line 73 can be changed to a link to the README for installation and configuration. The rest of the needed operations can be documented here:
1. Install and config
2. Plugging the dongle
3. Starting the agent (for details, refer [here](https://github.com/canonical/openthread-border-router-snap#run))

## How to Set Up and Control a Matter Thread Lighting Device

This guide will walk you through the process of building, running, and controlling a Matter Thread lighting application over a Thread network. 

Here are step-by-step instructions:

#### Tested Hardware
- Machine A
  - nRF52840 dongle as [Radio Co-Processor (RCP)](https://openthread.io/platforms/co-processor#radio_co-processor_rcp)
  - PC running Ubuntu Desktop 23.04 OS
- Machine B
  - nRF52840 dongle as RCP
  - PC running Ubuntu Desktop 23.04 OS
 
#### Environment Setup
- Machine A running
  - [OpenThread Border Router (OTBR) snap](https://snapcraft.io/openthread-border-router)
  - Matter Thread ligting App
- Machine B running
  - OTBR snap
  - [Chip Tool snap](https://snapcraft.io/chip-tool)

#### Versions Tested in This Guide
- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)

> - RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)

Author: The version should actually refers to the openthread SDK, not the Matter SDK. Reference links have been added in the latest commit.

#### 1. Form a Thread Network on Machine A
On Machine A, following the guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 

> using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 

Reviewer: ```suggestion
using the nRF52840 Dongle as the RCP. For details on setting up the OpenThread RCP, 
```

> please refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).

Reviewer: Redundant text.
```suggestion
please refer [here](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```

#### 2. Run a Matter Thread Lighting App on Machine B

> Build and run the [Matter Thread lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. By doing so, you will be able to observe the lighting status (on/off) from the logs later.

Reviewer: ```suggestion
Build and run the [Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. By doing so, you will be able to observe the lighting status (on/off) from the logs later.
```
